cadence/contracts/PrizeSavings.cdc

```diff
@@ -1005,16 +1820,34 @@ access(all) contract PrizeSavings {
             panic("No request to pop")
         }
         
+        /// Returns the captured weight snapshot.
         access(all) view fun getTimeWeightedStakes(): {UInt64: UFix64} {
             return self.timeWeightedStakes
         }
     }
     
+    // ============================================================
+    // WINNER SELECTION TYPES
+    // ============================================================
+    
+    /// Result of a winner selection operation.
+    /// Contains parallel arrays of winners, their prize amounts, and NFT assignments.
     access(all) struct WinnerSelectionResult {
+        /// Array of winner receiverIDs.
         access(all) let winners: [UInt64]
+        
+        /// Array of prize amounts (parallel to winners array).
         access(all) let amounts: [UFix64]
+        
+        /// Array of NFT ID arrays (parallel to winners array).
+        /// Each winner can receive multiple NFTs.
         access(all) let nftIDs: [[UInt64]]
         
+        /// Creates a WinnerSelectionResult.
+        /// All arrays must have the same length.
+        /// @param winners - Array of winner receiverIDs
+        /// @param amounts - Array of prize amounts per winner
+        /// @param nftIDs - Array of NFT ID arrays per winner
         init(winners: [UInt64], amounts: [UFix64], nftIDs: [[UInt64]]) {
             pre {
                 winners.length == amounts.length: "Winners and amounts must have same length"
```

Comment:
"Winners and amounts must have same length but winners length is \(winners.length) and amounts length is \(amounts.length)"